tests/test_path.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+from typing import Callable
+
+import pytest
+
+from ggshield.path import generate_files_from_paths
+
+
+@pytest.mark.parametrize(
+    ["filename", "input_content", "expected_content"],
+    [
+        ("normal.txt", "Normal", "Normal"),
+        ("0-inside.txt", "Ins\0de", "Ins de"),
+    ],
+)
+def test_generate_files_from_paths(
+    tmp_path, filename: str, input_content: str, expected_content: str
+):
+    """
+    GIVEN a file
+    WHEN calling generate_files_from_paths() on it
+    THEN it returns the expected File instance
+    AND the content of the File instance is what is expected
+    """
+    path = tmp_path / filename
+    Path(path).write_text(input_content)
+
+    files = list(generate_files_from_paths([str(path)], verbose=False))
+
+    file = files[0]
+    assert file.filename == str(path)
+    assert file.document == expected_content
+
+    assert len(files) == 1
+
+
+@pytest.mark.parametrize(
+    ["filename", "creator"],
+    [
+        ("archive.tar", lambda x: x.write_text("Uninteresting")),
```

Comment:
Perhaps we could name the file "a_binary_file.tar" to make the ignore reason even more explicit